services-camel/src/test/java/edu/unc/lib/cdr/ReplicationProcessorIT.java

```diff
@@ -0,0 +1,193 @@
+import java.net.URI;
+
+import org.apache.activemq.util.ByteArrayInputStream;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.apache.http.HttpStatus;
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoOperationFailedException;
+import org.fcrepo.client.FcrepoResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.mockito.Mockito.when;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.Repository;
+import edu.unc.lib.dl.fcrepo4.RepositoryPathConstants;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.util.URIUtil;
+
+/**
+ * @author harring
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/spring-test/test-fedora-container.xml","/spring-test/cdr-client-container.xml"})
+public class ReplicationProcessorIT extends CamelTestSupport {
+    private static final String MIMETYPE = "text/plain";
+
+    @Autowired
+    protected String baseAddress;
+
+    @Autowired
+    protected FcrepoClient client;
+    
+    @Autowired
+    protected Repository repository;
+
+    ReplicationProcessor processor;
+
+    @Mock
+    Exchange exchange;
+    @Mock
+    Message message;
+
+    @Before
+    public void init() throws IOException {
+        PIDs.setRepository(repository);
+        processor = new ReplicationProcessor(repository, "/tmp", 3, 100L);
+        initMocks(this);
+        
+        when(exchange.getIn()).thenReturn(message);
+        when(message.getHeader(CdrBinaryPath)).thenReturn("path/to/bin");
+        when(message.getHeader(CdrBinaryMimeType)).thenReturn(MIMETYPE);
+        when(exchange.getOut()).thenReturn(message);
+    }
+
+    private URI createBaseContainer(String name) throws IOException, FcrepoOperationFailedException {
+        URI baseUri = URI.create(URIUtil.join(baseAddress, name));
+        // Create a parent object to put the binary into
+        try (FcrepoResponse response = client.put(baseUri).perform()) {
+            return response.getLocation();
+        } catch(FcrepoOperationFailedException e) {
+            if (e.getStatusCode() != HttpStatus.SC_CONFLICT) {
+                throw e;
+            }
+            // Ignore duplicate creation of base container
+            return baseUri;
+        }
+    }
+    
+    private URI determineRepositoryPath(URI baseUri) throws IOException, FcrepoOperationFailedException {
+    	    PID parentPid = null;
+        try (FcrepoResponse response = client.post(baseUri).perform()) {
+            parentPid = PIDs.get(response.getLocation());
```

Comment:
Seems like the PID bit isn't needed since its returning the uri (as a string), which should be the same thing as the response.getLocation()